pkg/storage/unified/sql/backend.go

```diff
@@ -214,28 +215,46 @@ func (b *backend) initPruner(ctx context.Context) error {
 		MaxWait:    time.Minute * 5,
 		ProcessHandler: func(ctx context.Context, key pruningKey) error {
 			return b.db.WithTx(ctx, ReadCommitted, func(ctx context.Context, tx db.Tx) error {
-				res, err := dbutil.Exec(ctx, tx, sqlResourceHistoryPrune, &sqlPruneHistoryRequest{
-					SQLTemplate:  sqltemplate.New(b.dialect),
-					HistoryLimit: defaultPrunerHistoryLimit,
+				req := &sqlPruneHistoryRequest{
```

Comment:
@JohnnyQQQQ I don't see any existing tests for this behavior -- do we have any and I am missing it, or has it just been tested manually?